api/app/signals/apps/signals/models/category_assignment.py

```diff
@@ -17,6 +18,16 @@ class CategoryAssignment(CreatedUpdatedModel):
 
     extra_properties = JSONField(null=True)  # TODO: candidate for removal
 
+    # Note on the deadline fields: nulls are allowed for historic data because
+    # historic service level objectives are not generally available.
```

Comment:
Perhaps add a bit more about the reason these fields are added and why/how they work?

Reply:
Added a giant comment with extra explanation here.